### ISSUE TITLE
`Qibo` cannot be installed on an M1 Mac

### ISSUE THREAD

**Issue by AlbertMitjans** [qiboteam/qibo#820]
**Describe the bug**
Running `pip install qibo` on an M1 Mac raises the following error:
```
note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> nlopt

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Additional context**
This problem arises due to the dependency with the `nlopt` package ([see link](https://github.com/DanielBok/nlopt-python/issues/13)).

I realised that this package is only used in one place: https://github.com/qiboteam/qibo/blob/e70426cdbb67dd4dbdf530b69681ad517254cfcf/src/qibo/noise_model.py#L333

It would be nice to consider making `nlopt` an optional dependency.

**Comment by scarrazza** [comment]
Thanks @AlbertMitjans, we have already discussed about the possibility to drop nlopt in #814. 
@AlejandroSopena suggestions?

**Comment by AlejandroSopena** [comment]
Just half an hour ago I tried to install qibo on an M1 Mac and encountered the same issue. I was planning to mention it tomorrow at the meeting. As a temporary solution, I removed nlopt from the requirements to install it. I use nlopt because it has the DIRECT optimization method that works well for the problem of learning noise. I just found out that this method is already included in the latest stable version of scipy, so we can use it from there. As soon as I can (I'll try to do it today), I'll make the change.
